WP.com/me tweaks for Gravatar

wellyshen opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to p8Slzc-3CG-p2

## Proposed Changes

* Updating the tooltip text under the avatar photo
* Removing the "Hide my photo and Gravatar profile" toggle (and clear the relevant code)
* Adding a profile description + co-branding logo for Gravatar

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* Please refer to p8Slzc-3CG-p2

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* [Setup Calypso](https://github.com/Automattic/wp-calypso?tab=readme-ov-file#getting-started)
* Run `yarn start`, and visit http://calypso.localhost:3000/me
* The text of the tooltip has been updated:

<img width="983" alt="截圖 2024-07-29 下午3 43 18" src="https://github.com/user-attachments/assets/2de5fd6d-4bd1-45fd-a7e3-c070aff247a9">

* The "Hide my photo and Gravatar profile" toggle has been removed:

<img width="649" alt="截圖 2024-07-29 下午3 47 00" src="https://github.com/user-attachments/assets/54bbad9c-66f5-4359-99df-4897775e852f">

* The Gravatar blurb and co-branding logo has been added:

| Desktop | Mobile |
| - | - |
| <img width="1020" alt="截圖 2024-07-29 下午4 08 19" src="https://github.com/user-attachments/assets/60cad6b2-ebb6-4b13-a204-1b2b051761cf"> | <img width="390" alt="截圖 2024-07-29 下午3 50 49" src="https://github.com/user-attachments/assets/e45c44e6-0039-452d-856d-a95b80e3ccca"> |

* Click the "visit your Gravatar profile" of the Gravatar blurb, a Gravatar profile page should be opened in a new tab

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

wellyshen committed:
WP.com/me tweaks for Gravatar

wellyshen committed:
Format text

wellyshen committed:
Remove locale from the gravatar.com URL

wellyshen committed:
Merge branch 'trunk' into update/wpocm-me-tweak-for-gravatar

wellyshen committed:
Add logo & Adjust styles & Encode email URL parameter

wellyshen committed:
Add WPAndGravatarLogo component

wellyshen commented:
@danielbachhuber Could you please review this PR? Thank you!

aaronfc commented on client/blocks/edit-gravatar/index.jsx:
```diff
@@ -276,11 +276,8 @@ export class EditGravatar extends Component {
 					</p>
 					<InfoPopover className="edit-gravatar__pop-over" position="left">
```
Minor thing but, can we add `text-align: justify` to the contents of this popup? I think it looks better:
![image](https://github.com/user-attachments/assets/716baa21-4d71-4e5b-8603-94ef15e26b2e)
vs
![image](https://github.com/user-attachments/assets/f1fd1774-1709-44a5-985d-f6944e91f6cf)

wellyshen replied:
I tried to add the `text-align: justify`, but the spacing between each word looks strange. Therefore, I ended up not adding the property.

aaronfc replied:
In the screenshot I made above I think it looks ok. But I guess it's a matter of opinion :). anyway, not a blocker!

aaronfc reviewed (approved):
Looks good and works well! 🚀

Aurorum commented on client/me/profile/index.jsx:
```diff
@@ -121,18 +127,31 @@ class Profile extends Component {
+										components: {
+											strong: <strong />,
+											button: (
+												<button
+													className="profile__gravatar-profile-button"
+													onClick={ this.openGravatarQuickEditor }
+												/>
```
I'm a bit concerned about the decision to use the button here - is it necessary? An ordinary link seems more appropriate and would save you a bit of styling. It would also avoid the current issue with how the text breaks weirdly.

<img width="191" alt="Screenshot 2024-07-29 at 12 13 19" src="https://github.com/user-attachments/assets/2798469b-3422-4b50-b770-2cb06065a147">

If a button is necessary, you could use the `Button` component and the `link` prop to also avoid the additional styles.

wellyshen replied:
@Aurorum Thank you for pointing out the issue, it has been fixed through [this commit](https://github.com/Automattic/wp-calypso/pull/93023/commits/63485738683e8cc9d6430f7fe24ee8c7ffac6ebf).

Aurorum reviewed (changes requested):
Hey @wellyshen! Just a quick thought below and one other (blocking) issue: when I click the "Visit your Gravatar profile" button, the saving state is immediately activated even though I haven't made any changes. This prevents me from making any further changes to my profile without reloading. 

<img width="1088" alt="Screenshot 2024-07-29 at 12 17 41" src="https://github.com/user-attachments/assets/9b3a74fe-db84-4c9c-8577-fe389203ae2c">

wellyshen committed:
Merge branch 'trunk' into update/wpocm-me-tweak-for-gravatar

danielbachhuber reviewed (commented):
A few bits of feedback...

![CleanShot 2024-07-29 at 04 59 40@2x](https://github.com/user-attachments/assets/b675257f-52d4-441d-bed9-a57d29ac16cd)

This text is a bit small and difficult to read. Should we increase the size?

Also, should it say "Your WordPress.com profile..." to better connect WordPress.com and Gravatar?

![CleanShot 2024-07-29 at 05 00 38@2x](https://github.com/user-attachments/assets/35539645-781d-42ab-843f-cd6a7b8587bd)

This connect screen is not pretty. I don't think we need to solve it right now, but it could use a fresh coat of paint.

![CleanShot 2024-07-29 at 05 03 00@2x](https://github.com/user-attachments/assets/25f0be58-616f-4dba-9a20-34a5080d9c9b)

This is a pretty poor UX too. If we don't have a small window-optimized version of the Gravatar editor, then I think we should open it in a new tab.

![CleanShot 2024-07-29 at 05 04 51@2x](https://github.com/user-attachments/assets/dcb73d57-9c43-44f5-a725-def16b770837)

I edited my display name in Gravatar but it didn't seem to sync back to WordPress.com

![CleanShot 2024-07-29 at 05 06 20@2x](https://github.com/user-attachments/assets/66cd4a41-8926-463e-842c-586faeab70de)

Similarly, I added a link in my WordPress.com profile but it didn't get synced over to Gravatar.

wellyshen committed:
Address feedback

wellyshen commented:
> This text is a bit small and difficult to read. Should we increase the size?

✅ The text size has been increased to `14px`

> Also, should it say "Your WordPress.com profile..." to better connect WordPress.com and Gravatar?

✅ I have revised the wording

> This connect screen is not pretty. I don't think we need to solve it right now, but it could use a fresh coat of paint.

We have a GitHub issue (108374-gh-Automattic/gravatar) to customize and enhance the screen design soon

> This is a pretty poor UX too. If we don't have a small window-optimized version of the Gravatar editor, then I think we should open it in a new tab.
> I edited my display name in Gravatar but it didn't seem to sync back to WordPress.com
> Similarly, I added a link in my WordPress.com profile but it didn't get synced over to Gravatar.

We currently don't have a mechanism for mutual data synchronization, which would require additional technical integration on both the backend and frontend. To avoid any potential UX confusion at this stage, we're opening the Gravatar profile page in a new tab instead. (cc @burtrw)

@danielbachhuber Thank you for the review. I have addressed your feedback in [this commit](https://github.com/Automattic/wp-calypso/pull/93023/commits/63485738683e8cc9d6430f7fe24ee8c7ffac6ebf). Please take another look 👀.

Aurorum commented on client/me/profile/style.scss:
```diff
@@ -41,6 +42,19 @@
+
+	@include breakpoint-deprecated(">960px") {
+		flex-direction: row;
+		gap: 60px;
```
Such a tiny nit but I'd personally reduce this to 50px so that the link isn't on its own line.

wellyshen replied:
@Aurorum Thank you for the feedback. I have addressed it through [this commit](https://github.com/Automattic/wp-calypso/pull/93023/commits/a58353e765266b9a34099adc6fae155248f36899).

Aurorum commented on client/me/profile/index.jsx:
```diff
@@ -121,18 +113,26 @@ class Profile extends Component {
+							<span>
+								{ this.props.translate(
+									'Your WordPress.com profile is powered by Gravatar. Your Gravatar is public by default and may appear on any site using Gravatar when you’re logged in with {{strong}}%(email)s{{/strong}}.' +
+										' To manage your profile and privacy settings, {{a}}visit your Gravatar profile here{{/a}}.',
```
And other final tiny nit, I'd either remove the word "here" (ie. "visit your Gravatar profile") or I'd put the link on "here". It seems a bit unnecessary when the link is on the whole string. 

(You might also want to add a String Freeze label so that this can get some translations before merging)

wellyshen replied:
@Aurorum Thank you for the feedback. I have addressed it through [this commit](https://github.com/Automattic/wp-calypso/pull/93023/commits/a58353e765266b9a34099adc6fae155248f36899).

I will add the label, thanks :)

Aurorum reviewed (approved):
This works and looks much better - thanks for addressing the feedback! :)

danielbachhuber commented on client/me/profile/index.jsx:
```diff
@@ -121,18 +113,26 @@ class Profile extends Component {
+						<p className="profile__gravatar-profile-description">
+							<span>
+								{ this.props.translate(
+									'Your WordPress.com profile is powered by Gravatar. Your Gravatar is public by default and may appear on any site using Gravatar when you’re logged in with {{strong}}%(email)s{{/strong}}.' +
```
```suggestion
									'Your WordPress.com profile is connected to Gravatar. Your Gravatar is public by default and may appear on any site using Gravatar when you’re logged in with {{strong}}%(email)s{{/strong}}.' +
```

Given so few fields actually sync with Gravatar, I think it's better to say "connected to" over "powered by"

wellyshen replied:
Thank you for the feedback. I have addressed it through [this commit](https://github.com/Automattic/wp-calypso/pull/93023/commits/a58353e765266b9a34099adc6fae155248f36899).

danielbachhuber commented on client/me/profile/index.jsx:
```diff
@@ -121,18 +113,26 @@ class Profile extends Component {
+							<span>
+								{ this.props.translate(
+									'Your WordPress.com profile is powered by Gravatar. Your Gravatar is public by default and may appear on any site using Gravatar when you’re logged in with {{strong}}%(email)s{{/strong}}.' +
+										' To manage your profile and privacy settings, {{a}}visit your Gravatar profile here{{/a}}.',
```
```suggestion
										' To manage your Gravatar profile and visibility settings, {{a}}visit your Gravatar profile here{{/a}}.',
```

wellyshen replied:
✅

danielbachhuber reviewed (commented):
> We currently don't have a mechanism for mutual data synchronization, which would require additional technical integration on both the backend and frontend. To avoid any potential UX confusion at this stage, we're opening the Gravatar profile page in a new tab instead.

It still seems less than ideal, but I'm no longer strongly opposed to this PR. Where does mutual data synchronization fit within priorities?

One other thing: we'll probably want to have these strings translated before we merge.

wellyshen committed:
Revise Gravatar profile description

wellyshen commented:
> It still seems less than ideal, but I'm no longer strongly opposed to this PR. Where does mutual data synchronization fit within priorities?

@danielbachhuber (cc @burtrw) Regarding the data sync between WP.com and Gravatar, we haven't prioritized it yet. However, here's how we can achieve that:

- Integrate [the Gravatar Quick Editor SDK](https://github.com/Automattic/gravatar/tree/trunk/web/packages/quick-editor#gravatar-quick-editor) into Calypso.
- The SDK provides [an onProfileUpdated event](https://github.com/Automattic/gravatar/tree/trunk/web/packages/quick-editor#quickeditorcoreoptions-type), allowing us to re-fetch the necessary data through the existing `me/settings` API endpoint (However, I recommend using a dedicated API endpoint specifically designed for this data).
- Here is the mobile design for the Gravatar Quick Editor (only the necessary feature items are shown). If the UX isn't satisfactory, we will need to make adjustments.

<img width="402" alt="截圖 2024-07-29 晚上10 56 00" src="https://github.com/user-attachments/assets/6d51f098-222f-4ce3-bde8-0b2371d1f5e1">

> One other thing: we'll probably want to have these strings translated before we merge.

I will add the label when the PR is ready, thanks :)

wellyshen committed:
Merge branch 'trunk' into update/wpocm-me-tweak-for-gravatar

wellyshen committed:
Merge branch 'trunk' into update/wpocm-me-tweak-for-gravatar

wellyshen commented:
FYI. These are the updated wordings:

<img width="1148" alt="截圖 2024-07-31 凌晨2 09 36" src="https://github.com/user-attachments/assets/3ae95084-c4a2-44fc-bcfe-7801f58e8ca3">

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16438086

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @wellyshen for including a screenshot in the description! This is really helpful for our translators.

danielbachhuber reviewed (approved):
[This implementation](https://github.com/Automattic/wp-calypso/pull/93023#issuecomment-2258926457) seems appropriate for now.

Thanks for addressing my feedback, @wellyshen ! I'm excited to see Gravatar continue to grow 😊

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16438086).

wellyshen committed:
Merge trunk
